connector/subgraph/src/DisputableVoting.ts

```diff
@@ -0,0 +1,213 @@
+    voteData.value1, // nays
+    voteData.value2, // totalPower
+    settingsId,
+    votingApp.PCT_BASE()
+  )
+  proposal.save()
+
+  populateVoteCollateralData(proposal, event)
+}
+
+export function handleCastVote(event: CastVoteEvent): void {
+  updateVoteState(event.address, event.params.voteId)
+
+  const voter = loadOrcreateSupporter(event.params.voter)
+
+  const votingApp = VotingContract.bind(event.address)
+  const proposal = getProposalEntity(event.address, event.params.voteId)
+  const miniMeToken = ERC20Contract.bind(votingApp.token())
+
+  const stake = miniMeToken.balanceOfAt(
+    event.params.voter,
+    proposal.snapshotBlock
+  )
+
+  const castVote = loadOrCreateCastVote(
+    event.address,
+    event.params.voteId,
+    event.params.voter
+  )
+  castVote.entity = voter.id
+  castVote.stake = stake
+  castVote.supports = event.params.supports
+  castVote.createdAt = event.block.timestamp
+  castVote.caster = event.params.caster
+  castVote.save()
+}
+
+export function handlePauseVote(event: PauseVoteEvent): void {
+  updateVoteState(event.address, event.params.voteId)
+  const votingApp = VotingContract.bind(event.address)
+  const agreementApp = AgreementContract.bind(votingApp.getAgreement())
+  const challengeData = agreementApp.getChallenge(event.params.challengeId)
+  const proposal = getProposalEntity(event.address, event.params.voteId)
+  proposal.challenger = challengeData.value1
+  proposal.challengeId = event.params.challengeId
+  proposal.challengeEndDate = challengeData.value2
+  proposal.save()
+}
+
+export function handleResumeVote(event: ResumeVoteEvent): void {
+  updateVoteState(event.address, event.params.voteId)
+}
+
+export function handleCancelVote(event: CancelVoteEvent): void {
+  updateVoteState(event.address, event.params.voteId)
+}
+
+export function handleExecuteVote(event: ExecuteVoteEvent): void {
+  updateVoteState(event.address, event.params.voteId)
+
+  const proposal = getProposalEntity(event.address, event.params.voteId)
+  proposal.executedAt = event.block.timestamp
+  proposal.save()
+}
+
+export function handleQuietEndingExtendVote(
+  event: QuietEndingExtendVoteEvent
+): void {
+  updateVoteState(event.address, event.params.voteId)
+}
+
+export function handleChangeRepresentative(
+  event: ChangeRepresentativeEvent
+): void {
+  const voter = loadOrcreateSupporter(event.params.voter)
+  voter.representative = event.params.representative
+  voter.save()
+}
+
+export function updateVoteState(votingAddress: Address, voteId: BigInt): void {
+  const votingApp = VotingContract.bind(votingAddress)
+  const voteData = votingApp.getVote(voteId)
+
+  const proposal = getProposalEntity(votingAddress, voteId)
+  proposal.yeas = voteData.value0
+  proposal.nays = voteData.value1
+  proposal.status =
+    proposal.status == 'Settled'
+      ? STATUS_SETTLED
+      : castVoteStatus(voteData.value5)
```

Comment:
Wondering if we should set `STATUS_SETTLED` as default on `castVoteStatus` instead of `Unknown` ? 

Same for `castVoteStatusNum` below

Reply:
not sure ,  why we would want to default to settled?